### PR TITLE
Modify query and store to show subclasses

### PR DESCRIPTION
**Pull request by guergana**
first commit: Modify store to accept subclasses
second commit: modify the query.
third commit: merge
fourth commit: this field should be unchecked by default and should be checked true automatically if and only if the user selects a wikibase-item property

### CODE REVIEW

File: tests/unit/sparql/buildQuery.spec.ts [hunk 1]
```diff
@@ -162,10 +172,35 @@ describe( 'buildQuery', () => {
...
 		],
 		omitLabels: false,
 		} );
 		expect( actualQuery.replace( /\s+/g, ' ' ) ).toEqual( expectedQuery.replace( /\s+/g, ' ' ) );
 	} );
+
+	it( 'builds a query from a property and a string value with subclasses', () => {
+		const propertyId = 'P666';
+		const value = 'blah';
+		const subclassesId = 'P279';
```
Comment (guergana) [inline tests/unit/sparql/buildQuery.spec.ts]: How do we pass the environment variable here? I tried `const subclassesId = process.env.VUE_APP_SUBCLASS_PROPERTY;` and it didnt work

Comment (micgro42) [reply tests/unit/sparql/buildQuery.spec.ts]: You could try 

```
process.env = Object.assign(process.env, { VUE_APP_SUBCLASS_PROPERTY: 'value' });
```

( which I found there: https://github.com/vuejs/vue-test-utils/issues/193#issuecomment-438437626 )

Comment (guergana) [reply tests/unit/sparql/buildQuery.spec.ts]: Uf!! That was hard. The proposed solution wasn't working but this one worked: https://stackoverflow.com/a/58953365/1619792

File: package-lock.json [hunk 2]
```diff
@@ -4194,19 +4194,19 @@
       "dependencies": {
         "ansi-regex": {
           "version": "2.1.1",
-          "resolved": "https://registry.npmjs.org/ansi-regex/-/ansi-regex-2.1.1.tgz",
+          "resolved": "https://registry.npm.taobao.org/ansi-regex/download/ansi-regex-2.1.1.tgz",
```
Comment (micgro42) [inline package-lock.json]: We don't want those changes.

File: src/components/QueryCondition.vue [hunk 3]
```diff
@@ -91,9 +91,10 @@ export default Vue.extend( {
 				if ( selectedProperty === null ) {
 					this.$store.dispatch( 'unsetProperty', this.conditionIndex );
 					return;
+				} else if ( selectedProperty.datatype === 'wikibase-item' ) {
```
Comment (micgro42) [inline src/components/QueryCondition.vue]: I don't think we need this `else` as the previous `if` always returns

File: src/components/QueryCondition.vue [hunk 4]
```diff
@@ -91,9 +91,10 @@ export default Vue.extend( {
...
+					this.$store.dispatch( 'setSubclasses', { conditionIndex: this.conditionIndex, subclasses: true } );
```
Comment (micgro42) [inline src/components/QueryCondition.vue:96]: We do need to set this back to false when we select a different property, right?

File: package.json [hunk 5]
```diff
@@ -77,5 +77,10 @@
   "repository": {
     "type": "git",
     "url": "git+https://github.com/wmde/query-builder.git"
+  },
+  "jest": {
+    "setupFiles": [
+      "./tests/unit/setEnvVars.js"
```
Comment (micgro42) [inline package.json]: Should that not be `.jest/setEnvVars.js`? I'm confused that the tests are still passing :thinking:

File: tests/unit/store/mutations.spec.ts [hunk 6]
```diff
@@ -100,12 +103,40 @@ describe( 'mutations', () => {
...
+				conditionId: '0.123',
+				subclasses: true,
+			} ],
+			limit: 0,
+			useLimit: false,
+			omitLabels: true,
+			errors: [],
+		};
+
+		mutations.setSubclasses( state, { subclasses: expectedValue, conditionIndex: 0 } );
```
Comment (micgro42) [inline tests/unit/store/mutations.spec.ts:128]: This test sets the subclasses from `true` to `true`. Either the old or the new value should be `false` to actually test something.

**Review by micgro42 — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 2 above)
Comment (guergana) [reply package-lock.json]: oh yes, missed that one

File: src/components/QueryCondition.vue (hunk 3 above)
Comment (guergana) [reply src/components/QueryCondition.vue]: I wasn't sure if we needed it either.

File: package.json (hunk 5 above)
Comment (guergana) [reply package.json]: Actually this was from another way I tried that didn't work. So this is actually not being used. Thanks for pointing it out.

File: src/components/QueryCondition.vue (hunk 4 above)
Comment (guergana) [reply src/components/QueryCondition.vue:96]: I added an else here

File: tests/unit/store/mutations.spec.ts (hunk 6 above)
Comment (guergana) [reply tests/unit/store/mutations.spec.ts:128]: ah, yes, good catch.

File: package-lock.json [hunk 7]
```diff
@@ -4200,13 +4200,13 @@
         },
         "ansi-styles": {
           "version": "2.2.1",
-          "resolved": "https://registry.npmjs.org/ansi-styles/-/ansi-styles-2.2.1.tgz",
+          "resolved": "https://registry.npm.taobao.org/ansi-styles/download/ansi-styles-2.2.1.tgz",
```
Comment (micgro42) [inline package-lock.json]: No, I mean, we don't want to have this file checked in at all for this PR (and also do not want to use registry.npm.taobao.org)

File: src/components/QueryCondition.vue (hunk 4 above)
Comment (micgro42) [reply src/components/QueryCondition.vue:96]: :+1:

File: tests/unit/components/QueryCondition.spec.ts [hunk 8]
```diff
@@ -102,6 +102,26 @@ describe( 'QueryCondition.vue', () => {
 		expect( store.dispatch ).toHaveBeenCalledWith( 'updateValue', { value: userInput, conditionIndex } );
 	} );
 
+	it( 'set subclasses to true when property type is wikibase-item', async () => {
```
Comment (micgro42) [inline tests/unit/components/QueryCondition.spec.ts:105]: Can we have a test for setting it to false as well if the datatype is `string`?

**Review by micgro42 — CHANGES_REQUESTED** [review]
Thanks! We are almost there :slightly_smiling_face:

File: package-lock.json (hunk 7 above)
Comment (guergana) [reply package-lock.json]: oh, weird. I thought I removed this.

**Review by micgro42 — APPROVED** [review]

**Pull request merged**